### PR TITLE
ENH: Added serialization for IncrementalPCA

### PR DESCRIPTION
**Pull request by olegkkruglov**
### Description 
* Contains changes from https://github.com/intel/scikit-learn-intelex/pull/1924 which are necessary for correctly working multiple times serialization/deserialization
* Added `__getstate__` and `__setstate__` methods to `IncrementalPCA`
* Finalization is called with serialization during `__getstate__` method
* `_need_to_finalize` flag is added to avoid unnecessary call of finalization backend.
* Added test for checking if it works well
* Updated deselected_tests.yaml

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes, if necessary.
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

File: onedal/decomposition/tests/test_incremental_pca.py [hunk 1]
```diff
@@ -196,3 +196,62 @@ def test_on_random_data(
         whiten and queue is not None and queue.sycl_device.device_type.name == "gpu"
     ):
         assert_allclose(transformed_data, expected_transformed_data, atol=tol)
+
+
+@pytest.mark.parametrize("queue", get_queues())
+@pytest.mark.parametrize("dtype", [np.float32])
```
Comment (samir-nasibli) [inline onedal/decomposition/tests/test_incremental_pca.py:202]: Why only for float32?

File: sklearnex/preview/decomposition/tests/test_incremental_pca.py [hunk 2]
```diff
@@ -264,3 +264,46 @@ def test_sklearnex_partial_fit_on_random_data(
     X_df = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
     transformed_data = incpca.transform(X_df)
     check_pca(incpca, dtype, whiten, X, transformed_data)
+
+
+@pytest.mark.parametrize("dataframe,queue", get_dataframes_and_queues())
+@pytest.mark.parametrize("dtype", [np.float32])
```
Comment (samir-nasibli) [inline sklearnex/preview/decomposition/tests/test_incremental_pca.py:270]: The same comment for the dtype testing. Make sense extend for float64

**Comment by samir-nasibli** [comment]
@olegkkruglov is this PR on development? what is the status for it?

**Comment by icfaust** [comment]
Should have changes from #1925 and should be quick to validate/merge.

**Review by icfaust — APPROVED** [review]

File: onedal/decomposition/tests/test_incremental_pca.py [hunk 3]
```diff
@@ -196,3 +196,66 @@ def test_on_random_data(
...
+    gen = np.random.default_rng(seed)
+    X = gen.uniform(low=-0.3, high=+0.7, size=(10, 10))
+    X = X.astype(dtype)
+    X_split = np.array_split(X, 2)
+    incpca.partial_fit(X_split[0], queue=queue)
+    incpca_loaded.partial_fit(X_split[0], queue=queue)
+    assert incpca._need_to_finalize == True
+    assert incpca_loaded._need_to_finalize == True
+
+    # Check that estmator can be serialized after partial_fit call.
```
Comment (icfaust) [inline onedal/decomposition/tests/test_incremental_pca.py:223]:
```suggestion
    # Check that estimator can be serialized after partial_fit call.
```

**Review by icfaust — COMMENTED** [review]
May be worth making changes suggested in #2180 before merging this (including the spelling mistake, and the check of other parameters).

File: onedal/decomposition/pca.cpp [hunk 4]
```diff
@@ -15,6 +15,8 @@
 *******************************************************************************/
 #include "oneapi/dal/algo/pca.hpp"
 #include "onedal/common.hpp"
+#define NO_IMPORT_ARRAY // import_array called in table.cpp
```
Comment (samir-nasibli) [inline onedal/decomposition/pca.cpp:18]: Please update description for this addition
